core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -702,7 +704,9 @@ void core_mmu_set_user_map(struct core_mmu_user_map *map)
 
 enum core_mmu_fault core_mmu_get_fault_type(uint32_t fault_descr)
 {
-	assert(fault_descr & FSR_LPAE);
+	if ((fault_descr & FSR_LPAE) == 0)
```

Comment:
An `assert()` should be enough here, this is more of a configuration issue than a run-time issue (even if it's detect run-time).